src/component-library/NumberInput/NumberInput.tsx

```diff
@@ -1,39 +1,58 @@
 import * as React from 'react';
+
 import { BaseNumberInput } from './NumberInput.style';
-import { InputProps } from 'component-library/Input';
+import { useNumberField } from '@react-aria/numberfield';
+import { useNumberFieldState } from '@react-stately/numberfield';
+
+type Props = { value?: number; defaultValue?: number; step?: number; label?: string; description?: string };
+
+const defaultProps = {};
 
-// `onWheel` prop can't be used with `preventDefault` because
-// React implements passive event listeners.
-const disableChangeOnWheel = (event: MouseEvent) => {
-  event.preventDefault();
-};
+type NativeAttrs = Omit<React.InputHTMLAttributes<Element>, keyof Props>;
 
-type NumberInputProps = InputProps;
+type NumberInputProps = Props & NativeAttrs & Partial<typeof defaultProps>;
 
 const NumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>(
   (props, ref): JSX.Element => {
-    const inputParent = React.useRef<HTMLDivElement | null>(null);
+    const { disabled, readOnly, required, onChange, ...rest } = props;
+
+    const ariaProps = {
+      isDisabled: disabled,
+      isReadOnly: readOnly,
+      isRequired: required,
+      ...rest
+    };
+
+    const state = useNumberFieldState({ ...ariaProps, locale: 'en-US' });
+    const inputRef = React.useRef<HTMLInputElement>(null);
+
+    React.useImperativeHandle(ref, () => inputRef.current as any);
```

Comment:
What's the reason for using `useImperativeHandle` here?

Reply:
Im using it because this is a forwardRef component. Since my component makes use a local inputRef, the `useImperativeHandle` makes the forwardRef "point" to my local inputRef. Think of it like a merge.

Reply:
Are there any other ways to handle this? Not a big deal if not, but if there isn't then we'll probably want to add some documentation here (if that's the case we should probably move it to a utility, since I assume this is a pattern we'll be using a lot, and would be good to document it in one place).

Reply:
I will create an util for this.